### DISCUSSION TITLE
Error when generating world

### DISCUSSION THREAD

**Discussion by ican9595** [FieldRobotEvent/Virtual_Field_Robot_Event#30] [Q&A] [answered]
Hi,

I have cloned the latest version of this repository and installed everything successfully so far.
When I try to create a simulation world by running `rosrun virtual_maize_field create_task_1_mini.sh` or any other `create_task_` I get the following output:

> QObject::moveToThread: Current thread (0x1ce2b00) is not the object's thread (0x27c3c70).
Cannot move to target thread (0x1ce2b00)

> qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "/usr/local/lib/python3.6/dist-packages/cv2/qt/plugins" even though it was found.

> This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

> Available platform plugins are: xcb, eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, webgl.

> /home/ican/my_catkin_ws/src/Virtual_Field_Robot_Event/virtual_maize_field/scripts/create_task_1.sh: Zeile 8: 24579 Abgebrochen             (Speicherabzug geschrieben) rosrun virtual_maize_field generate_world.py --row_length 10 --rows_left 0 --rows_right 11 --rows_curve_budget 0.78539816339 --row_segments straight,curved --row_segment_curved_radius_min 4.0 --row_segment_curved_radius_max 5.0 

Is this error related to the package or to my own linux setup? Unfortunately I wasn't able to fix it but as I said I have followed the installation steps one by one, installed all the dependencies etc.

Help would be appreciated.

Thank you very much!

**Comment by Darkproduct** [comment]
I have to check this. I'm not sure why there is any Qt dependency there. I added cv2 for the heightmap generation, but Qt should not be needed.

**Comment by Darkproduct** [comment, marked as answer]
Maybe thats your problem:

In 6422adeaec986b12729bb843ec28cd40517fff6f
The packes you need are in the package.xml:
```
python3-rospkg
python3-jinja2
python3-opencv
python3-matplotlib
python3-shapely
```
